model/mod_list.py

```diff
@@ -208,5 +326,61 @@ def focusOutEvent(self, e: QFocusEvent) -> None:
         return super().focusOutEvent(e)
 
     def mod_clicked(self, item: QListWidgetItem) -> None:
-        """Method to handle clicking on a row"""
-        self.mod_info_signal.emit(item.data(Qt.UserRole)["packageId"])
+        """
+        Method to handle clicking on a row
+        Look up the mod's data by uuid
+        """
+        self.mod_info_signal.emit(item.data(Qt.UserRole)["uuid"])
+
+    def get_mod_url(self, widget_json_data) -> str:
+        url = ""
+        if (
+            widget_json_data["data_source"] == "workshop"
+        ):  # If the mod was parsed from a Steam mods source
+            if widget_json_data.get("steam_url") and isinstance(
+                widget_json_data["steam_url"], str
+            ):  # If the steam_url exists
+                url = widget_json_data.get("steam_url")  # Use the Steam URL
+            elif widget_json_data.get("url") and isinstance(
+                widget_json_data["url"],
+                str,
+            ):  # Otherwise, check if local metadata url exists
+                url = widget_json_data["url"]  # Use the local metadata url
+            else:  # Otherwise, warn & do nothing
+                logger.warning(f"Unable to get url for mod {widget_package_id}") # TODO: Make warning visible
+        elif (
+            widget_json_data["data_source"] == "local"
+        ):  # If the mod was parsed from a local mods source
+            if widget_json_data.get("url") and isinstance(
+                widget_json_data["url"],
+                str,
+            ):  # If the local metadata url exists
+                url = widget_json_data["url"]  # Use the local metadata url
+            elif widget_json_data.get("steam_url") and isinstance(
```

Comment:
https://github.com/oceancabbage/RimSort/blob/sprint-a1.0.3/util/mods.py#L243

The "steam_url" is dynamically added if a PublishedFileId.txt is parsed and a pfid is added. It is really a fallback in the event that an About.xml doesn't have a "url" k/v, there will at least be a Steam publishing to fall back on.

Most mod developers I see are putting a Github repo as their About.xml "url" - however I think this is a good solution so that users have something given what is available.

It will always use "url" over "steam_url" - and if there is a PFID, the "Open mod in steam" option is also dynamically added to the contextMenu

Reply:
I suppose I could also try to make it based on whether or not there even is one of those values. If it doesn't have url or steam_url, we could just make it not show the menu at all.

Reply:
Additional comments added. 

Fixed